test/unit/spec/IDPDiscovery_spec.js

```diff
@@ -0,0 +1,1031 @@
+          spyOn(test.router, 'navigate');
+          test.form.helpFooter().click();
+          test.form.forgotPasswordLink().click();
+          expect(test.router.navigate).toHaveBeenCalledWith('signin/forgot-password', {trigger: true});
+        });
+      });
+      itp('does not navigate to forgot password page when link disabled and clicked', function () {
+        return setup().then(function (test) {
+          spyOn(test.router, 'navigate');
+          test.form.setUsername('testuser@clouditude.net');
+          test.setNextWebfingerResponse(resSuccess);
+          test.form.submit();
+          return tick(test);
+        }).then(function(test) {
+          test.form.helpFooter().click();
+          test.form.forgotPasswordLink().click();
+          expect(test.router.navigate).not.toHaveBeenCalledWith('signin/forgot-password', {trigger: true});
+        });
+      });
+      itp('navigates to custom forgot password page when available', function () {
+        return setup({ 'helpLinks.forgotPassword': 'https://foo.com' }).then(function (test) {
+          spyOn(SharedUtil, 'redirect');
+          test.form.helpFooter().click();
+          test.form.forgotPasswordLink().click();
+          expect(SharedUtil.redirect).toHaveBeenCalledWith('https://foo.com');
+        });
+      });
+      itp('does not navigate to custom forgot password page when link disabled and clicked', function () {
+        return setup({ 'helpLinks.forgotPassword': 'https://foo.com' }).then(function (test) {
+          spyOn(SharedUtil, 'redirect');
+          test.form.setUsername('testuser@clouditude.net');
+          test.setNextWebfingerResponse(resSuccess);
+          test.form.submit();
+          return tick(test);
+        }).then(function(test) {
+          test.form.helpFooter().click();
+          test.form.forgotPasswordLink().click();
+          expect(SharedUtil.redirect).not.toHaveBeenCalledWith('https://foo.com');
+        });
+      });
+      itp('unlock link is hidden on load', function () {
+        return setup({'features.selfServiceUnlock': true}).then(function (test) {
+          expect(test.form.unlockLinkVisible()).toBe(false);
+        });
+      });
+      itp('shows unlock link when clicking help', function () {
+        return setup({'features.selfServiceUnlock': true}).then(function (test) {
+          test.form.helpFooter().click();
+          expect(test.form.unlockLinkVisible()).toBe(true);
+        });
+      });
+      itp('navigates to unlock page when click unlock link', function () {
+        return setup({'features.selfServiceUnlock': true}).then(function (test) {
+          spyOn(test.router, 'navigate');
+          test.form.helpFooter().click();
+          test.form.unlockLink().click();
+          expect(test.router.navigate).toHaveBeenCalledWith('signin/unlock', {trigger: true});
+        });
+      });
+      itp('does not navigate to unlock page when link disabled and clicked', function () {
```

Comment:
same here. The link is disabled, but if we have a success response, then we don;t have the link at all, right?

Reply:
Same explanation as for the forgot password link